app/models/charge.js

```diff
@@ -86,15 +110,21 @@ module.exports = function (correlationId) {
     })
   }
 
-  function findWithEventsError (err, response, reject) {
+  function getStatusCodeForError (err, response) {
+    let status = 'CLIENT_UNAVAILABLE'
     const code = (response || {}).statusCode || (err || {}).errorCode
-    if (code === 404) return reject('NOT_FOUND')
-    if (code > 200) return reject('GET_FAILED')
-    reject('CLIENT_UNAVAILABLE')
+    if (code > 200) status = 'GET_FAILED'
+    if (code === 404) status = 'NOT_FOUND'
```

Comment:
This line will never be reached, because `404` will be caught by the first condition `if (code > 200)`.
That's why they've been in the previous order.

Reply:
A `let` can be overridden - this isn't a terminal operation

Reply:
my bad while reading.
I still think it would make more sense to return from the if statement (and it would require the change of the order back to the previous one).